### PR TITLE
Altar Functionality

### LINKED ISSUE #232: make altar gameobject

### LINKED ISSUE #237: Implement altar functionality
interactable by player
when used generate new dungeon floor
keep old core to keep hp, mana, cards, items
teleport player to new spawn point
increment dungeon score

### PR DESCRIPTION
**Pull request by SebastienGorman**
resolves #237 
resolves #232 

Still missing : Replace grey cylinder with actual altar asset.

SampleScene won't be supported with this feature. If you need it to be, ping me on slack.

### CODE REVIEW

**Review by RossBaumgartner — APPROVED** [review]
tested and working ✔

File: Assets/Scripts/LevelGenerator/Altar.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+using UnityEngine.SceneManagement;
+
+public class Altar : MonoBehaviour, IInteractable
+{
+    public void Interact()
+    {
+        Debug.Log("Player found the altar!");
+        UIManager.MyInstance.IncrementFinalRoomScore();
+        DungeonInitializer.MyInstance.DestroyAndRepopulate();
+        //Scene scene = SceneManager.GetActiveScene(); // Not the best implementation for when we will want to have saves working, but it works for now.
```
Comment (AndreiBaetica) [inline Assets/Scripts/LevelGenerator/Altar.cs:15]: can probably delete these comments

File: Assets/Scripts/LevelGenerator/DungeonInitializer.cs [hunk 2]
```diff
@@ -7,12 +7,36 @@
...
+    public static DungeonInitializer instance;
+    
+    void Awake()   
+    {
+        instance = this;
+    }
+        
+    #endregion
+    
+    public static DungeonInitializer MyInstance
```
Comment (AndreiBaetica) [inline Assets/Scripts/LevelGenerator/DungeonInitializer.cs:21]: can we rename this to something more descriptive? Perhaps DungeonInstance

File: Assets/Scripts/Controllers/ActorController.cs [hunk 3]
```diff
@@ -227,6 +227,8 @@ protected virtual void Die()
         GameObject enemyDropGameObject = (GameObject) Instantiate(enemyDrop,
             transform.position,
             Quaternion.identity);
+        Transform spawnableParent = GameObject.Find("Spawnables").transform;
```
Comment (AndreiBaetica) [inline Assets/Scripts/Controllers/ActorController.cs:230]: This same code shows up several times (playerController, Spawners). Could we just put it in Start() of this class and make it accessible through a spawnablesParent variable?

File: Assets/Scripts/Managers/GameLoopManager.cs [hunk 4]
```diff
@@ -42,23 +42,33 @@ public static void SetPlayerTurn(bool playerTurn)
...
+                    if (!activeChars[i].enabled) activeChars.RemoveAt(i);
+                    else if (!activeChars[i].doneTurn) allDone = false;
+                }
+            }
+            
+            if (allDone)
             {
-                activeChar.doneTurn = false;
+                reservedPositions.Clear();
+                //Debug.Log("setting player turn true");
```
Comment (AndreiBaetica) [inline Assets/Scripts/Managers/GameLoopManager.cs:60]: can remove this comment

**Pull request merged**